src/software/power/charger.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <pins.h>
+
+/*
+ * Represents the charger on the power board
+ */
+class Charger
+{
+   public:
+    /**
+     * Creates a Charger setting up pins and attaching interrupts.
+     */
+    Charger();
+    /**
+     * Sets the charge mode with a given operation. Valid operations are LOW and HIGH.
+     *
+     * @param op operation to be set
+     */
+    void setChargeMode(int op);
+    /**
+     * Sets up a callback for when charging is done. This callback is only called once
+     *
+     * @param charge_done_callback callback function to set
+     */
+    void setChargeDoneCallbackOnce(void (*charge_done_callback)());
+    /**
+     * Returns the voltage of the capacitors
+     *
+     * @return voltage of capacitors
+     */
+    float getCapacitorVoltage();
+    /**
+     * Returns the status of the flyback fault
+     *
+     * @return whether the flyback fault is tripped or not
+     */
+    bool getFlybackFault();
+
+   private:
+    /**
+     * Isr called when charging is done. This checks the flyback status and also calls the
+     * charge_done_callback.
+     */
+    static void IRAM_ATTR chargeDone();
```

Comment:
what is `IRAM_ATTR` ?

Reply:
It tells the compiler to put the function into embedded ram so we can call the function faster than if it was stored in flash and since chargeDone is run as an interrupt service routine we want to run it as fast as possible.  More info: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/memory-types.html#iram-instruction-ram